geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/DestroyJndiBindingFunction.java

```diff
@@ -36,9 +36,12 @@ public CliFunctionResult executeFunction(FunctionContext<Object[]> context) {
     if (destroyingDataSource) {
       typeName = "Data source";
       DataSource dataSource = lookUpDataSource(jndiName);
-      if(dataSource != null && !(dataSource.getClass() == GemFireBasicDataSource.class) && !(dataSource.getClass() == GemFireConnPooledDataSource.class)) {
+      if (dataSource != null && !(dataSource.getClass() == GemFireBasicDataSource.class)
```

Comment:
I think this expression could be cleaned up a bit.
use instanceof instead of .getClass() ==. Also with instanceof no need for a null check.
So you could have a method like this:
private boolean wasCreatedAsDataSource(DataSource dataSource) {
  return dataSource instanceof GemFireBasicDataSource
    || dataSource instanceof GemFireConnPoolDataSource;
}
The in this code you could just say:
  if (!wasCreatedAsDataSource(dataSource) {
    error...

Reply:
Agreed, we should switch to instance of to remove usage of == and to improve code reliability.